osu.Game/Overlays/NowPlayingOverlay.cs

```diff
@@ -308,6 +346,10 @@ private void trackChanged(WorkingBeatmap beatmap, TrackChangeDirection direction
                         BeatmapMetadata metadata = beatmap.Metadata;
                         title.Text = new RomanisableString(metadata.TitleUnicode, metadata.Title);
                         artist.Text = new RomanisableString(metadata.ArtistUnicode, metadata.Artist);
+                        if (title.Text.ToString().Length > 30)
+                        {
+                            title.Text = title.Text.ToString().Substring(0, 27) + "...";
```

Comment:
Please use https://github.com/ppy/osu-framework/blob/2d4142c5433492868f1ba7a2f670afeb84a9e906/osu.Framework/Graphics/Sprites/SpriteText.cs#L254-261

Also why is the title special? The artist text could also overflow.

Reply:
Oh, and also for the record: calling `.ToString()` here destroys the functionality of `RomanisableString` (i.e. being able to display metadata in original language immediately after the corresponding setting changes)